### PR TITLE
Add TestKit to BOM and archetype modules [ECR-3056]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Add TestKit to BOM and archetype modules.

---
See: https://jira.bf.local/browse/ECR-3056

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -97,6 +98,11 @@
       <version>${mockito.version}</version>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>com.exonum.binding</groupId>
+      <artifactId>exonum-testkit</artifactId>
```
Comment (dmitry-timofeev) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/pom.xml]: test scope?

**Comment by dmitry-timofeev** [comment]
The archetype project must depend on the testkit as well (as it does on core in runtime scope), as documented in the issue.

File: exonum-java-binding/service-archetype/pom.xml [hunk 2]
```diff
@@ -43,6 +51,11 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-archetype-plugin</artifactId>
           <version>${maven-archetype.version}</version>
+          <configuration>
+            <properties>
+              <nativeLibPath>${project.parent.basedir}/core/rust/target/debug</nativeLibPath>
```
Comment (dmitry-timofeev) [inline exonum-java-binding/service-archetype/pom.xml]: @dip56 Would you please test that release builds continue to work (we didn't use to use the native library in archetype ITs), and the generated project works with the installed app (the current 0.7.0-SNAPSHOT)?

File: exonum-java-binding/service-archetype/src/main/resources/archetype-resources/src/test/java/ServiceModuleTest.java [hunk 3]
```diff
@@ -45,6 +48,19 @@ void testTransactionConverterBinding() {
     assertThat(s, instanceOf(MyTransactionConverter.class));
   }
 
+  @Test
+  void testServiceInstantiation() {
```
Comment (bullet-tooth) [inline exonum-java-binding/service-archetype/src/main/resources/archetype-resources/src/test/java/ServiceModuleTest.java]: wouldn't it be better to create a separate class with junit5 extension?

Comment (dmitry-timofeev) [reply exonum-java-binding/service-archetype/src/main/resources/archetype-resources/src/test/java/ServiceModuleTest.java]: Not sure. This particular test is only useful at project initialization to verify the app is installed properly + as an example of testkit configuration :thinking: We can extend some comments here and there/modify the test display name to make their purpose more apparent.

Comment (dmitry-timofeev) [reply exonum-java-binding/service-archetype/src/main/resources/archetype-resources/src/test/java/ServiceModuleTest.java]: I removed the existing tests as they do not bring much value on top of this new one.

Comment (dmitry-timofeev) [reply exonum-java-binding/service-archetype/src/main/resources/archetype-resources/src/test/java/ServiceModuleTest.java]: and documented the likely failure

**Pull request merged**